packages/@core.keystone/plugins/softDeleted.js

```diff
@@ -78,11 +80,98 @@ const softDeleted = ({ deletedAtField = 'deletedAt', newIdField = 'newId' } = {}
     return { fields, hooks, access: newAccess, ...rest }
 }
 
-function applySoftDeletedFilters (access, deletedAtField) {
+/**
+ * Gets "where" variables from http graphQL request
+ * @param {Object} args
+ * @return {Object}
+ */
+function getWhereVariables (args) {
+    const method = get(args, ['context', 'req', 'method'])
+    let variables = {}
+    if (method === 'GET') {
+        variables = get(args, ['context', 'req', 'query', 'variables'])
+    }
+    else if (method === 'POST') {
+        variables = get(args, ['context', 'req', 'body', 'variables'])
+    }
+    try {
+        return JSON.parse(variables).where || {}
+    } catch (e) {
+        return {}
+    }
+}
+
+/**
+ * Checks if query has the soft deleted property on first level of depth
+ * @param {Object} whereQuery
+ * @param {string} deletedAtField
+ * @return {boolean}
+ */
+function queryHasSoftDeletedField (whereQuery, deletedAtField) {
+    return Object.keys(whereQuery).find((x) => x.startsWith(deletedAtField))
+}
+
+/**
+ * Checks if query has the soft deleted property on any level of depth
+ * @param {Object} whereQuery
+ * @param {string} deletedAtField
+ * @return {boolean}
+ */
+function queryHasSoftDeletedFieldDeep (whereQuery, deletedAtField) {
```

Comment:
Example: Filter employee by not deleted roles: `where = { role : { deletedAt: null } }` -- we don't want to remove deletedAt from Employee list

Reply:
Yep, good case, thank you. Need to rewrite the function

Reply:
This case is handled by explicitly setting the deletedAt field on all query parts, so I documeted it and created a tech debt task as well as todo